### ISSUE TITLE
get_room_options via XMLRPC is broken

### ISSUE THREAD

**Issue by encodeltd** [processone/ejabberd#982]
Consider the following test using `Ejabberd 16.02`:

```shell
$ ejabberdctl create_room test conference example.com
$ ejabberdctl get_room_options test conference
title   
description
allow_change_subj       true
allow_query_users       true
allow_private_messages  true
allow_private_messages_from_visitors    anyone
allow_visitor_status    true
allow_visitor_nickchange        true
public  true
public_list     true
persistent      false
moderated       true
captcha_protected       false
members_by_default      true
members_only    false
allow_user_invites      false
password_protected      false
password
anonymous       true
presence_broadcast      [moderator,participant,visitor]
allow_voice_requests    true
voice_request_min_interval      1800
max_users       200
logging false
vcard   
captcha_whitelist       {0,nil}
mam     false
$ ejabberdctl destroy_room test conference
$ ejabberdctl get_room_options test conference
$ 
```

Now, let's test the same via XMLRPC.

Here is `test_get_room_options.py`:

```python
import xmlrpclib

server_url = 'http://127.0.0.1:4560'
# server = xmlrpclib.ServerProxy(server_url, verbose=True)
server = xmlrpclib.ServerProxy(server_url, verbose=False)

EJABBERD_XMLRPC_LOGIN = {'user': 'admin',
                         'password': 'admin',
                         'server': 'example.com',
                         'admin': True}

def ejabberdctl(command, data):
    fn = getattr(server, command)
    return fn(EJABBERD_XMLRPC_LOGIN, data)

print "create room test"
print ejabberdctl('create_room', {'name': 'test', 'service': 'conference', 'host': 'example.com'})

try:
    print "try to get_room_options"
    print ejabberdctl('get_room_options', {'name': 'test', 'service': 'conference'})
except:
    print "get_room_options is broken"

print "destroy room test"
print ejabberdctl('destroy_room', {'name': 'test', 'service': 'conference'})

print "get_room_options now works and returns {'options': []} as expected"
print ejabberdctl('get_room_options', {'name': 'test', 'service': 'conference'})

print "create room test again"
print ejabberdctl('create_room', {'name': 'test', 'service': 'conference', 'host': 'example.com'})

print "let the get_room_options fail so we see the exception thrown at us"
print ejabberdctl('get_room_options', {'name': 'test', 'service': 'conference'})
```

Here is the test result:

```shell
$ python test_get_room_options.py 
create room test
{'res': 1}
try to get_room_options
get_room_options is broken
destroy room test
{'res': 0}
get_room_options now works and returns {'options': []} as expected
{'options': []}
create room test again
{'res': 0}
let the get_room_options fail so we see the exception thrown at us
Traceback (most recent call last):
  File "test_get_room_options.py", line 35, in <module>
    print ejabberdctl('get_room_options', {'name': 'test', 'service': 'conference'})
  File "test_get_room_options.py", line 14, in ejabberdctl
    return fn(EJABBERD_XMLRPC_LOGIN, data)
  File "/usr/lib/python2.7/xmlrpclib.py", line 1233, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib/python2.7/xmlrpclib.py", line 1591, in __request
    verbose=self.__verbose
  File "/usr/lib/python2.7/xmlrpclib.py", line 1273, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib/python2.7/xmlrpclib.py", line 1303, in single_request
    response = h.getresponse(buffering=True)
  File "/usr/lib/python2.7/httplib.py", line 1073, in getresponse
    response.begin()
  File "/usr/lib/python2.7/httplib.py", line 415, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python2.7/httplib.py", line 379, in _read_status
    raise BadStatusLine(line)
httplib.BadStatusLine: ''
```

Relevant bit from `/var/log/ejabberd/ejabberd.log`:
```
2016-03-05 08:16:18.426 [info] <0.470.0>@ejabberd_listener:accept:333 (#Port<0.18127>) Accepted connection 127.0.0.1:53305 -> 127.0.0.1:4560
2016-03-05 08:16:18.426 [debug] <0.1455.0>@ejabberd_http:init:154 S: [{[],ejabberd_xmlrpc}]
2016-03-05 08:16:18.426 [info] <0.1455.0>@ejabberd_http:init:158 started: {gen_tcp,#Port<0.18127>}
2016-03-05 08:16:18.426 [debug] <0.1455.0>@ejabberd_http:process_header:281 (#Port<0.18127>) http query: 'POST' <<"/RPC2">>
2016-03-05 08:16:18.427 [debug] <0.1455.0>@ejabberd_http:extract_path_query:395 client data: <<"<?xml version='1.0'?>\n<methodCall>\n<methodName>get_room_options</methodName>\n<params>\n<param>\n<value><struct>\n<member>\n<name>admin</name>\n<value><boolean>1</boolean></value>\n</member>\n<member>\n<name>password</name>\n<value><string>admin</string></value>\n</member>\n<member>\n<name>user</name>\n<value><string>admin</string></value>\n</member>\n<member>\n<name>server</name>\n<value><string>example.com</string></value>\n</member>\n</struct></value>\n</param>\n<param>\n<value><struct>\n<member>\n<name>name</name>\n<value><string>test</string></value>\n</member>\n<member>\n<name>service</name>\n<value><string>conference</string></value>\n</member>\n</struct></value>\n</param>\n</params>\n</methodCall>\n">>
2016-03-05 08:16:18.427 [debug] <0.1455.0>@ejabberd_http:process:353 [<<"RPC2">>] matches []
2016-03-05 08:16:18.427 [debug] <0.1455.0>@ejabberd_xmlrpc:process:245 got XML-RPC request: {call,get_room_options,[{struct,[{admin,true},{password,<<"admin">>},{user,<<"admin">>},{server,<<"example.com">>}]},{struct,[{name,<<"test">>},{service,<<"conference">>}]}]}
2016-03-05 08:16:18.427 [debug] <0.1455.0>@ejabberd_commands:execute_command2:409 Executing command mod_muc_admin:get_room_options with Args=[<<"test">>,<<"conference">>]
2016-03-05 08:16:18.432 [error] <0.1455.0> CRASH REPORT Process <0.1455.0> with 0 neighbours crashed with reason: bad argument in call to erlang:iolist_to_binary([moderator,participant,visitor]) in fxmlrpc:encode_param/1 line 120
```

According to the `CRASH REPORT` it's the `presence_broadcast`'s  value`[moderator,participant,visitor]` that breaks `get_room_options` when called via XMLRPC.

**Comment by badlop** [comment]
Thanks! I hope it's fixed now. Please reopen if it still crashes.

**Comment by encodeltd** [comment]
Thank you! That worked. I applied the fix, rebuilt 16.02 and retested.

```python
import xmlrpclib

server_url = 'http://127.0.0.1:4560'
server = xmlrpclib.ServerProxy(server_url, verbose=False)

EJABBERD_XMLRPC_LOGIN = {'user': 'admin',
                         'password': 'admin',
                         'server': 'example.com',
                         'admin': True}

def ejabberdctl(command, data):
    fn = getattr(server, command)
    return fn(EJABBERD_XMLRPC_LOGIN, data)

print "create room"
print ejabberdctl('create_room', {'name': 'test',
                                  'service': 'conference',
                                  'host': 'example.com'})

print "get room options"
print ejabberdctl('get_room_options', {'name': 'test',
                                       'service': 'conference'})
```

```shell
$ python test_get_room_options.py 
create room
{'res': 1}
get room options
{'options': [{'option': [{'name': 'title'}, {'value': ''}]}, {'option': [{'name': 'description'}, {'value': ''}]}, {'option': [{'name': 'allow_change_subj'}, {'value': 'true'}]}, {'option': [{'name': 'allow_query_users'}, {'value': 'true'}]}, {'option': [{'name': 'allow_private_messages'}, {'value': 'true'}]}, {'option': [{'name': 'allow_private_messages_from_visitors'}, {'value': 'anyone'}]}, {'option': [{'name': 'allow_visitor_status'}, {'value': 'true'}]}, {'option': [{'name': 'allow_visitor_nickchange'}, {'value': 'true'}]}, {'option': [{'name': 'public'}, {'value': 'true'}]}, {'option': [{'name': 'public_list'}, {'value': 'true'}]}, {'option': [{'name': 'persistent'}, {'value': 'false'}]}, {'option': [{'name': 'moderated'}, {'value': 'true'}]}, {'option': [{'name': 'captcha_protected'}, {'value': 'false'}]}, {'option': [{'name': 'members_by_default'}, {'value': 'true'}]}, {'option': [{'name': 'members_only'}, {'value': 'false'}]}, {'option': [{'name': 'allow_user_invites'}, {'value': 'false'}]}, {'option': [{'name': 'password_protected'}, {'value': 'false'}]}, {'option': [{'name': 'password'}, {'value': ''}]}, {'option': [{'name': 'anonymous'}, {'value': 'true'}]}, {'option': [{'name': 'presence_broadcast'}, {'value': '[moderator,participant,visitor]'}]}, {'option': [{'name': 'allow_voice_requests'}, {'value': 'true'}]}, {'option': [{'name': 'voice_request_min_interval'}, {'value': '1800'}]}, {'option': [{'name': 'max_users'}, {'value': '200'}]}, {'option': [{'name': 'logging'}, {'value': 'false'}]}, {'option': [{'name': 'vcard'}, {'value': ''}]}, {'option': [{'name': 'captcha_whitelist'}, {'value': '{0,nil}'}]}, {'option': [{'name': 'mam'}, {'value': 'false'}]}]}
```
